DOCS-4888 Added clarification about resource-based transports.

Cristian-Venticinque opened the pull request:
Added clarification about resource-based transports.

Commit:
Added clarification about resource-based transports.

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -109,6 +109,12 @@ All three basic types of transports are supported in clusters in different ways,
++
+[NOTE]
+====
+Have in mind that Mule only coordinates that different nodes in the cluster don’t try to access the same resource.
```
I don't think you need the first sentence in the Note since it is explained above the note. Suggested rewrite of the second sentence.:

If you are running Mule applications or non-Mule applications outside the cluster that need to access resources within the cluster, you must implement logic to lock those resources.

Cristian-Venticinque replied:
Thanks, I agree that the first sentence is redundant. Regarding the second, I'm not sure if "resources within the cluster" is accurate because I think the resources could be located outside the cluster as well. 
The context that needs to be covered here is when: _non-Mule apps or Mule apps outside the cluster need to access the same resources as the nodes within a cluster_

dhirsch1 replied:
That works.

dhirsch1 replied:
I thought you might want to put Mule apps outside the cluster before non-Mule apps. Whatever is most common I guess.

Cristian-Venticinque replied:
I ordered it that way because non-Mule apps will always be outside of the cluster, while Mule apps can be in or out, that's why the clarification affects directly Mule apps.

Commit:
Clarified note about about Resource-based transport.

Commit:
Created sections for each transport type. Reorganized information.

-Information about transport types was scattered, now each section 
contains all info related to that transport type.
